File: data/src/main/java/dev/olog/data/api/deezer/DeezerAlbumResponse.java
package dev.olog.data.api.deezer;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class DeezerAlbumResponse {

    public List<Data> data = new ArrayList<>();

    public static class Data {

        public String cover;

        @SerializedName("cover_big")
        public String coverBig;

        @SerializedName("cover_medium")
        public String coverMedium;

        @SerializedName("cover_small")
        public String coverSmall;

        @SerializedName("cover_xl")
        public String coverXl;

    }

}

File: data/src/main/java/dev/olog/data/api/deezer/DeezerAlbumResponse.kt
package dev.olog.data.api.deezer

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class DeezerAlbumResponse(
    val data: List<Data>? = null,
) {

    @Serializable
    class Data(
        val cover: String? = null,
        @SerialName("cover_big")
        val coverBig: String? = null,
        @SerialName("cover_medium")
        val coverMedium: String? = null,
        @SerialName("cover_small")
        val coverSmall: String? = null,
        @SerialName("cover_xl")
        val coverXl: String? = null,
    )
}